### PR TITLE
Add an example for radio interferometric imaging in astronomy

### PR DESCRIPTION
**Pull request by ChaoTang0330**
### Checks to be done before submitting your PR
- [x] `python3 -m pytest tests/` runs successfully.
- [x] `black .` runs successfully.
- [x] `make html` runs successfully (in the `docs/` directory).
- [x] Updated docstrings related to the changes (as applicable).
- [x] Added an entry to the [CHANGELOG.rst](../CHANGELOG.rst).

### CODE REVIEW

File: examples/advanced/demo_ri_basic.py [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+image_gdth = torch.from_numpy(image_gdth).unsqueeze(0).unsqueeze(0)
+
+
+def to_logimage(im, rescale=False, dr=5000):
+    r"""
+    A function plotting the image in logarithmic scale with specified dynamic range
+    """
+    if rescale:
+        im = im - im.min()
+        im = im / im.max()
```
Comment (tobias-liaudat) [inline examples/advanced/demo_ri_basic.py:118]: Shouldn't the rescaling be to [0,1]? 
For that, it should be something like `im = (im - im.min()) / (im.max() - im.min())`

Comment (ChaoTang0330) [reply examples/advanced/demo_ri_basic.py:118]: Yes, the rescaling is to [0,1] and the two ways are equivalent.

Comment (tobias-liaudat) [reply examples/advanced/demo_ri_basic.py:118]: Ah yes, my bad thanks @ChaoTang0330 :)

File: examples/advanced/demo_ri_basic.py [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+#
+#         physics = RadioInterferometry(img_size=img_size, samples_loc=samples_loc, device=device)
+#
+#
+# Groundtruth image
+# ----------------------------------------------------------------------------------------
+# The following data is our groundtruth with the settings of Experiment II in `Aghabiglou et al. (2024) <https://arxiv.org/abs/2403.05452>`_.
+# The groundtruth data has been normalized in the [0, 1] range.
+# As usual in radio interferometric imaging, the data has high dynamic range,
+# i.e. the ratio between the faintest and highest emissions is higher than in traditional low-level vision task.
```
Comment (tachella) [inline examples/advanced/demo_ri_basic.py:153]: task -> tasks.

**Review by tachella — CHANGES_REQUESTED** [review]
the PR looks great, I only found minor issues to fix.

regarding the docs of the RadioInterferometry class, I think we can reuse the nice mathematical explanation in the example.

File: deepinv/optim/prior.py [hunk 3]
```diff
@@ -333,6 +333,8 @@ class WaveletPrior(Prior):
     :math:`\Psi` is an orthonormal wavelet transform, and :math:`\|\cdot\|_{p}` is the :math:`p`-norm, with
     :math:`p=0`, :math:`p=1`, or :math:`p=\infty`.
 
+    If clamping parameters are provided, the prior writes as :math:`\reg{x} = \|\Psi x\|_{p} + \iota_{c_{\text{min}, c_{\text{max}}}`.
```
[the remark this answers is not in the document]
Comment (matthieutrs) [reply deepinv/optim/prior.py]: You are right I've updated

File: deepinv/models/wavdict.py [hunk 4]
```diff
@@ -371,3 +377,12 @@ def forward(self, y, ths=0.1):
             if rel_crit < 1e-3:
                 break
         return x
+
+    def psi(self, x, wavelet="db2", level=2, dimension=2):
+        r"""
+        Returns a flattened list containing the wavelet coefficients for each wavelet.
```
[the remark this answers is not in the document]
Comment (matthieutrs) [reply deepinv/models/wavdict.py:388]: Yes! Ive added it

File: CHANGELOG.rst [hunk 5]
```diff
@@ -167,4 +168,6 @@ Authors
 .. _Minh Hai Nguyen: https://fr.linkedin.com/in/minh-hai-nguyen-7120
 .. _Florian Sarron: https://fsarron.github.io/
 .. _Pierre Weiss: https://www.math.univ-toulouse.fr/~weiss/
-.. _Zhiyuan Hu: https://github.com/zhiyhu1605
\ No newline at end of file
+.. _Zhiyuan Hu: https://github.com/zhiyhu1605
+.. _Chao Tang:: https://github.com/ChaoTang0330
```
[the remark this answers is not in the document]
Comment (matthieutrs) [reply CHANGELOG.rst]: Fixed

File: deepinv/physics/radio.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    import torchkbnufft as tkbn
+except:
+    tkbn = ImportError("The torchkbnufft package is not installed.")
+
+
+class RadioInterferometry(LinearPhysics):
+    r"""
+    Radio Interferometry measurement operator.
+
+    The operator handles ungridded measurements using the NUFFT, which is based in Kaiser-Bessel kernel interpolation.
```
[the remark this answers is not in the document]
Comment (matthieutrs) [reply deepinv/physics/radio.py]: Indeed

File: deepinv/physics/radio.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+try:
+    import torchkbnufft as tkbn
+except:
+    tkbn = ImportError("The torchkbnufft package is not installed.")
+
+
+class RadioInterferometry(LinearPhysics):
+    r"""
+    Radio Interferometry measurement operator.
+
```
[the remark this answers is not in the document]
Comment (matthieutrs) [reply deepinv/physics/radio.py:13]: Added

File: deepinv/physics/radio.py [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    The operator handles ungridded measurements using the NUFFT, which is based in Kaiser-Bessel kernel interpolation.
+
+
+    :param tuple img_size: Size of the target image, e.g., (H, W).
+    :param torch.Tensor samples_loc: Normalized sampling locations in the Fourier domain (Size: N x 2).
+    :param torch.Tensor dataWeight: Data weighting for the measurements (Size: N). Default is `torch.tensor([1.0])`.
+    :param Union[int, Sequence[int]] interp_points: Number of neighbors to use for interpolation in each dimension. Default is `7`.
+    :param float k_oversampling: Oversampling of the k space grid, should be between `1.25` and `2`. Default is `2`.
+    :param bool real_projection: Apply real projection after the adjoint NUFFT. Warning: If the `real_projection` is `False`, the output of the adjoint will have a complex-typed rather than a float-typed.
```
[the remark this answers is not in the document]
Comment (matthieutrs) [reply deepinv/physics/radio.py]: Done

File: deepinv/utils/plotting.py [hunk 9]
```diff
@@ -191,12 +193,20 @@ def plot(
 
     for i, row_imgs in enumerate(imgs):
         for r, img in enumerate(row_imgs):
-            axs[r, i].imshow(img, cmap=cmap, interpolation=interpolation)
+            im = axs[r, i].imshow(img, cmap=cmap, interpolation=interpolation)
```
[the remark this answers is not in the document]
Comment (matthieutrs) [reply deepinv/utils/plotting.py:196]: This allows to have colorbars for each subplots

**Review by tachella — APPROVED** [review]
thanks for applying the modifications, the PR looks good to go!

**Pull request merged**